src/test/java/swinglib/JComboBoxBuilderTest.java

```diff
@@ -0,0 +1,125 @@
+package swinglib;
+
+import java.awt.event.ItemEvent;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.swing.JComboBox;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.Is;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import games.strategy.triplea.settings.ClientSetting;
+
+
+@RunWith(MockitoJUnitRunner.StrictStubs.class)
+public class JComboBoxBuilderTest {
+  @Mock
+  private ItemEvent mockItemEvent;
+
+  @Before
+  public void setUp() {}
+
+  @Test(expected = IllegalArgumentException.class)
+  public void buildInvalidEmptyOptionSet() {
+    JComboBoxBuilder.builder()
+        .menuOptions()
+        .build();
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void builderNoMenuOptionSpecified() {
+    JComboBoxBuilder.builder()
+        .build();
+  }
+
+  @Test
+  public void basicBuilderWithMenuOptions() {
+    final JComboBox box = JComboBoxBuilder.builder()
+        .menuOptions("option 1", "option 2", "option 3")
+        .build();
+
+    MatcherAssert.assertThat(box.getSelectedIndex(), Is.is(0));
+    MatcherAssert.assertThat(box.getItemCount(), Is.is(3));
+    MatcherAssert.assertThat(box.getItemAt(0), Is.is("option 1"));
+    MatcherAssert.assertThat(box.getItemAt(1), Is.is("option 2"));
+    MatcherAssert.assertThat(box.getItemAt(2), Is.is("option 3"));
+    MatcherAssert.assertThat(
+        // TODO: document which item listener is attached by default, why is this not zero?
+        box.getItemListeners().length, Is.is(1));
+    MatcherAssert.assertThat(box.getSelectedItem(), Is.is("option 1"));
+  }
+
+  @Test
+  public void itemListener() {
+
+    final AtomicInteger triggerCount = new AtomicInteger(0);
+
+    final String secondOption = "option 2";
+    final JComboBox box = JComboBoxBuilder.builder()
```

Comment:
Static analysis: "JComboBox is a raw type. References to generic type JComboBox<E> should be parameterized"